### PR TITLE
Fellowship type pages

### PR DESCRIPTION
**Pull request by mmmavis**
Related to #951 

Updated comps: https://projects.invisionapp.com/share/ZJ9YTAGEA#/screens/273312988

- fellowship_type.html template
- science fellowship page
- open web fellowship page

https://foundation-mofostaging--pr-994.herokuapp.com/fellowships/science/
https://foundation-mofostaging--pr-994.herokuapp.com/fellowships/open-web/

### CODE REVIEW

**Comment by mmmavis** [comment]
@alanmoo this is ready for review!

File: source/sass/utilities.scss [hunk 1]
```diff
@@ -31,3 +31,29 @@
   background-repeat: no-repeat;
   background-size: contain;
 }
+
+.breadcrumb {
```
Comment (mmmavis) [inline source/sass/utilities.scss]: overriding some bootstrap's rules...

File: network-api/networkapi/fellows/templates/fellows_openweb.html [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+{% block breadcrumb_fellowship_type %}
+Open Web
+{% endblock %}
+
+{% block fellowship_type %}
+Open Web
+{% endblock %}
+
+{% block fellowship_description %}
+{% lorem 2 p %}
```
Comment (alanmoo) [inline network-api/networkapi/fellows/templates/fellows_openweb.html]: Neat

File: source/js/main.js [hunk 3]
```diff
@@ -231,6 +232,42 @@ let main = {
...
+    // Science Fellowship
+    if (document.getElementById(`featured-science-fellow`)) {
+      let metadata = {
+        featured: true,
+        'internet_health_issues': [ `Decentralization`, `Open Innovation` ],
+        links: [],
+        name: `Firstname Surname`,
+        role: `[Area Fellowship] Fellow, [Year]`,
+        location: `City, Country`,
+        image: `https://images.pexels.com/photos/264206/pexels-photo-264206.jpeg?w=500`,
```
Comment (alanmoo) [inline source/js/main.js]: This is the secret gem of this PR

File: source/js/main.js [hunk 4]
```diff
@@ -231,6 +232,42 @@ let main = {
...
+        quote: `Quote quote quote quote quote quote quote quote quote quote quote quote.`,
+        affiliations: [ `Stanford University Professor; YouthLAB founder`],
+        'fellow_directory_link': { type: `science`, link: `/fellowships/directory` }
+      };
+
+      ReactDOM.render(<Person metadata={metadata} />, document.getElementById(`featured-open-web-fellow`));
+    }
```
Comment (alanmoo) [inline source/js/main.js]: This feels WETter than necessary...could the metadata be passed in as props on the target div (in the markup), and then render run on a generic class?

File: network-api/networkapi/templates/foundation-base.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <!-- This is a copy/paste of landing-page.html, modified to be useful for fellowships to start. Ideally the entire site will extend this template eventually -->
 <!DOCTYPE html>
-<html>{% load settings_value %}{% load active_nav %}
+<html>{% load settings_value %}{% load fellowship_active_nav %}
```
Comment (alanmoo) [inline network-api/networkapi/templates/foundation-base.html]: I realize I'm a bit concerned with editing of this file too much to be specific to fellows (and still calling it `foundation-base`). Perhaps we should abstract out a level so we can rely on a true "foundation base" generated from the original pug file?  I'd say for this PR, let's start with renaming this file.

Comment (mmmavis) [reply network-api/networkapi/templates/foundation-base.html]: k i'll rename it to `fellowships_base.html` as the first step

File: source/js/main.js [hunk 6]
```diff
@@ -231,6 +232,42 @@ let main = {
     if (document.querySelector(`#news`)) {
       ReactDOM.render(<News env={env}/>, document.querySelector(`#news`));
     }
+
+    // Science Fellowship
```
Comment (gvn) [inline source/js/main.js]: Can we put this into a new file? `main.js` is getting pretty large and this is fairly specific to just one section.

File: source/sass/type.scss [hunk 7]
```diff
@@ -41,6 +41,24 @@ a {
...
+    letter-spacing: 0.25px;
+  }
+
+  &:hover,
+  &#{$active-class-name} {
+    color: inherit;
+    text-decoration: none;
+    border-bottom: 5px solid #ffed00;
+    position: relative;
+    top: 2.5px;
```
Comment (gvn) [inline source/sass/type.scss]: All browsers round subpixels differently, so its better to just use integer values with pixel units IMO.

**Comment by mmmavis** [comment]
There are some CSS tweaks I need to apply. Will flag someone to review after I'm done.

File: source/js/fellowships.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Person from './components/people/person.jsx';
+
+// Embed various Fellowships pages related React components
+function injectReactComponents() {
+  // Science Fellowship
+  if (document.getElementById(`featured-science-fellow`)) {
+    let metadata = {
```
Comment (mmmavis) [inline source/js/fellowships.js]: @alanmoo 

We could feed these hardcoded metadata via HTML `data-*` attributes... but some of them aren't string. It's more troublesome to hardcode array and object as `data-*` values... and it'll make the HTML code less readable. I can't find a DRYer way to do this... at least not now as these values are hardcoded.

**Comment by mmmavis** [comment]
@alanmoo this is ready for review again!

File: source/js/fellowships.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      quote: `Quote quote quote quote quote quote quote quote quote quote quote quote.`,
+      affiliations: [`Stanford University Professor; YouthLAB founder`],
+      'fellow_directory_link': { type: `science`, link: `/fellowships/directory` }
+    };
+
+    ReactDOM.render(<Person metadata={metadata} />, document.getElementById(`featured-open-web-fellow`));
+  }
+}
+
+export default { injectReactComponents };
```
Comment (mmmavis) [inline source/js/fellowships.js]: Exporting a named function here. I think we'll end up having more functions in this module in the future.

**Comment by mmmavis** [comment]
@alanmoo re: nav

I haven't really worked on styling yet since mockups aren't finalized yet. But yea I can file a ticket to make sure we don't forget to make the nav responsive.

**Pull request merged**